src/site/antora/modules/ROOT/pages/manual/api.adoc

```diff
@@ -14,187 +14,366 @@
+public class MyApp {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    private static final Marker ACCOUNT_MARKER = MarkerManager.getMarker("ACCOUNT");
+
+    public void removeUser(String userId) {
+        logger.debug(ACCOUNT_MARKER, "Removing user with ID `{}`", userId);
+        // ...
+    }
+
 }
 ----
 
-With Java 8 you can achieve the same effect with a lambda expression.
-You no longer need to explicitly check the log level:
+xref:manual/markers.adoc[Read more on markers...]
+
+[#scoped-context]
+=== Scoped Context
+
+Just like a https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/ScopedValue.html[Java's `ScopedValue`], the visibility of tags are associated with the block they were introduced:
 
 [source,java]
 ----
-// Java-8 style optimization: no need to explicitly check the log level:
-// the lambda expression is not evaluated if the TRACE level is not enabled
-logger.trace("Some long-running operation returned {}", () -> expensiveOperation());
+
+private class Worker implements Runnable {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    public void authUser(Request request, Session session) {
+         // ... // <3>
+        ScopedContext
+                .where("ipAddress", request.getRemoteAddr()) // <1>
+                .where("hostName", request.getServerName()) // <1>
+                .where("loginId", session.getAttribute("loginId")) // <1>
+                .run(() -> { // <2>
+                    LOGGER.debug("Authenticating user");
+                    // ...
+                });
+         // ... // <3>
+    }
+
+}
 ----
+<1> Associating properties such that they will only be visible to Log4j **within the scope of the `run()` method**.
+These properties can later on be used to, for instance, filter the log event, provide extra information in the layout, etc.
+<2> The block determining the visibility scope of the provided properties.
+<3> Outside the scope of the `run()` method provided properties will not be visible.
 
-=== Logger Names
+xref:manual/scoped-context.adoc[Read more on Scoped Context...]
 
-Most logging implementations use a hierarchical scheme for matching
-logger names with logging configuration. In this scheme, the logger name
-hierarchy is represented by `'.'` characters in the logger name, in a
-fashion very similar to the hierarchy used for Java package names. For
-example, `org.apache.logging.appender` and `org.apache.logging.filter`
-both have `org.apache.logging` as their parent. In most cases,
-applications name their loggers by passing the current class's name to
-`LogManager.getLogger(...)`. Because this usage is so common, Log4j 2
-provides that as the default when the logger name parameter is either
-omitted or is null. For example, in all examples below the Logger will
-have a name of `"org.apache.test.MyTest"`.
+[#thread-context]
+=== Thread Context
```

Comment:
Should we split this one into "nested diagnostic context" and "mapped diagnostic context"?

Reply:
Tried to improve the section in cfc76c8c8566c9945ab7859cf5493b71c57a18c1, while trying to still keep it short, since:
1. Page is already pretty long
2. Thread Context is already explained elsewhere
3. We encourage users to use Scoped Context anyway